schema.production.kf

```diff
@@ -223,49 +199,47 @@ action get_wallets() public view {
+        GROUP BY address HAVING count(id) = 1)
+    THEN ERROR('A human needs to have at least one wallet. We can not delete the last wallet.')
+    END;
+
     DELETE FROM wallets
     WHERE id=$id AND human_id=(SELECT DISTINCT human_id FROM wallets WHERE (wallet_type = 'EVM' AND address=@caller COLLATE NOCASE)
         OR (wallet_type = 'NEAR' AND public_key = $converted)
     );
 }
 
-action has_profile($address) public view {
-    SELECT EXISTS (
-        SELECT 1 FROM wallets WHERE address=$address COLLATE NOCASE
-    ) AS has_profile;
-}
 
-@kgw(authn='true')
-action get_wallet_human_id() public view {
-    $converted = idos_near.implicit_address_to_public_key(@caller);
-    SELECT DISTINCT human_id FROM wallets
-    WHERE (wallet_type = 'EVM' AND address=@caller COLLATE NOCASE)
-        OR (wallet_type = 'NEAR' AND public_key = $converted);
-}
+// CREDENTIAL ACTIONS
 
-@kgw(authn='true')
-action get_credentials() public view {
-    $converted = idos_near.implicit_address_to_public_key(@caller);
-	SELECT DISTINCT c.id, c.human_id, c.issuer, c.credential_type, c.credential_level, c.credential_status, sc.original_id AS original_id
-    FROM credentials AS c
-    LEFT JOIN shared_credentials AS sc ON c.id = sc.duplicate_id
-    INNER JOIN wallets ON c.human_id = wallets.human_id
-    WHERE (
-        wallets.wallet_type = 'EVM' AND wallets.address = @caller COLLATE NOCASE
-    ) OR (
-        wallets.wallet_type = 'NEAR' AND wallets.public_key = $converted
-    );
+procedure upsert_credential_as_issuer(
+        $id uuid,
+        $human_id uuid,
+        $credential_type text,
+        $credential_level text,
+        $credential_status text,
+        $content text,
+        $encryption_public_key text) public returns (bool) {
+    $issuer text := get_issuer();
+    INSERT INTO credentials (id, human_id, credential_type, credential_level, credential_status, content, encryption_public_key, issuer)
+    VALUES ($id, $human_id, $credential_type, $credential_level, $credential_status, $content, $encryption_public_key, $issuer)
+    ON CONFLICT(id) DO UPDATE
+    SET credential_status=$credential_status, content=$content, encryption_public_key=$encryption_public_key, issuer=$issuer;
+
+    return true; // dummy return value is needed when the procedure is called from an action like `SELECT procedure_call()`;
 }
 
-action add_credential($id, $issuer, $credential_type, $credential_level, $credential_status, $content, $encryption_public_key) public {
+action add_credential($id, $credential_type, $credential_level, $credential_status, $content, $encryption_public_key) public {
     $converted = idos_near.implicit_address_to_public_key(@caller);
-    INSERT INTO credentials (id, human_id, issuer, credential_type, credential_level, credential_status, content, encryption_public_key)
+    INSERT INTO credentials (id, human_id, credential_type, credential_level, credential_status, content, encryption_public_key)
```

Comment:
I'm just noticing that we're now breaking away from the current semantics of this field. It used to mean "this issuer created the bytes in the `contents` field", and now it means "it was this issuer who inserted this row; if it's NULL, it was the user".

The first thing that comes to pkoch's mind is that we should make the team aware of this change (because it's going to be really surprising), and also make sure that the current integrators are ok with how this changed.

Alex pointed out that using `issuer` as a field name to signify "who inserted this row" can get really confusing for people who are looking at this with W3C VC naming in their minds.

Reply:
Perhaps we want to have a `credential_issuer` field to be what is today's `issuer` is, and to have a `inserting_party` field to keep track of who wrote what? :shrug:

Reply:
We keep the `issuer` field as it is, and introduce a new field (`inserting_party`, e.g.).

If this proves to be problematic, let's just eject this whole tracking functionality from the scope.